packages/@react-spectrum/numberfield/docs/NumberField.mdx

```diff
@@ -0,0 +1,320 @@
+used to choose whether to display the currency symbol, currency code, or currency name. Finally, the `currencySign` option
+can be set to `accounting` to use accounting notation for negative numbers, which uses parentheses rather than a minus sign
+in some locales.
+
+If you need to allow the user to change the currency, you should include a separate dropdown next to the NumberField.
+The NumberField itself will not determine the currency from the user input.
+
+```tsx example
+<NumberField
+  label="Transaction amount"
+  defaultValue={45}
+  formatOptions={{
+    style: 'currency',
+    currency: 'EUR',
+    currencyDisplay: 'code',
+    currencySign: 'accounting'
+  }} />
+```
+
+### Units
+
+The `style: 'unit'` option can be passed to the `formatOptions` prop to format the value with a unit of measurement. The `unit`
+option must also be passed to set which unit to use (e.g. `inch`). In addition, the `unitDisplay` option can be used to choose
+whether to display the unit in long, short, or narrow format.
+
+If you need to allow the user to change the unit, you should include a separate dropdown next to the number field.
+The number field itself will not determine the unit from the user input.
+
+**Note:** the unit style is not currently supported in Safari. A [polyfill](https://formatjs.io/docs/polyfills/intl-numberformat/)
+may be necessary.
+
+```tsx example
+<NumberField
+  label="Package width"
+  defaultValue={4}
+  formatOptions={{
+    style: 'unit',
+    unit: 'inch',
+    unitDisplay: 'long'
+  }} />
+```
+
+## Minimum and maximum values
+
+The `minValue` and `maxValue` props can be used to limit the entered value to a specific range. The value will be clamped
+when the user blurs the input field. In addition, the increment and decrement buttons will be disabled when the value is
+within one `step` value from the bounds ([see below](#step-values) for info about steps). Ranges can be open ended by only
+providing either `minValue` or `maxValue` rather than both.
+
+If a valid range is known ahead of time, it is a good idea to provide it to `NumberField` so it can optimize the experience.
+For example, when the minimum value is greater than or equal to zero, it is possible to use a numeric keyboard on iOS rather
+than a full text keyboard (necessary to enter a minus sign).
+
+```tsx example
+<NumberField
+  label="Enter your age"
+  minValue={0} />
+```
+
+## Step values
+
+The `step` prop can be used to snap the value to certain increments. If there is a `minValue` defined, the steps are calculated
+starting from the minimum. For example, if `minValue={2}`, and `step={3}`, the valid step values would be 2, 5, 8, 11, etc. If no
+`minValue` is defined, the steps are calculated starting from zero and extending in both directions. In other words, such that the
+values are evenly divisible by the step. If no `step` is defined, any decimal value may be typed, but incrementing and decrementing
```

Comment:
I'm trying to figure out my suggestion related to the last sentence, like do I want us to instead talk about stepping by decimal values. What my core though is, should we articulate steps being any positive number including numbers with decimals (3.14) or that are decimals (.33)?